Histogram equalization for color images

Arcadie opened the pull request:
This changes the logic of the processing for color images when histogram equalization is selected:

- perform 98% contrast adjustment
- colorize image
- histogram equalization

Everything else should remain the same.

I'm converting the image to [Lab](https://en.wikipedia.org/wiki/CIELAB_color_space), equalizing the histogram for the L value [0..100], and then converting back to RGBA.

Here is the difference between telemetry / histogram with false color

<details>
  <summary>Large image warning: 6MB</summary>
  
![telemetry-vs-historgam](https://user-images.githubusercontent.com/1472875/90965538-9e19a500-e4d1-11ea-860a-275adba9002e.gif)
  
</details>

Arcadie commented:
I've finally managed to make the histogram equalization work "in-place", without copying channels around.
This is more in line with the "processing pipeline" without all the re-assignments (`let img = do_something(&img)`).

I should have marked this as _draft_ when I started, in case you started reviewing it and I still made some changes, sorry about that.

One thing to be aware of are the `unsafe` blocks in histogram equalization. It's the same algorithm used in `imageproc::contrast::equalize_histogram_mut` [src](https://docs.rs/imageproc/0.21.0/src/imageproc/contrast.rs.html#164-183), but adapted to work with sub-images, as accessing the pixels is done differently. 

Equalizing the histogram for the L channel happens in a similar way, but in range [0..100], so vector length is 101.

Please let me know if you have any suggestions or concerns, thanks!

martinber commented on src/noaa_apt.rs:
```diff
@@ -150,28 +151,41 @@ pub fn process(
+    // then equalize histogram of color image if needed
+    if color.is_some() {
+        if let Contrast::Histogram = contrast_adjustment {
+            let (l, h) = misc::percent(&signal, 0.98)?;
+            low = l;
+            high = h;
```
```suggestion
            (low, high) = misc::percent(&signal, 0.98)?;
```

Arcadie replied:
Unfortunately destructuring assignments are not yet supported: https://github.com/rust-lang/rfcs/issues/372 so this is a workaround.

martinber reviewed (approved):
It's OK the way it is, these are some suggestions that I will do if I find time

martinber commented:
Very nice work. Sorry for taking so long. I will be very busy at least another week

Arcadie commented:
> Sorry for taking so long. I will be very busy at least another week

No worries! I've addressed all the issues, thanks for taking the time to look at this!

martinber commented:
Looks good!
